### PR TITLE
mercurial, tortoisehg: update to 5.6.1

### PR DESCRIPTION
**Pull request by danchr**
This also affects dependent ports. To make such upgrades easier, I converted git-remote-hg to use the Python port group while at it.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: python/py-mercurial_extension_utils/Portfile
Comment (reneeotten) [inline python/py-mercurial_extension_utils/Portfile:18]: what's the point of using the gitlab PG and then using PyPI as the `master_sites`? Is there a reason why you cannot pick one of the two approaches?

File: devel/hg-git/Portfile
Comment (lbschenkel) [inline devel/hg-git/Portfile:25]: Why pypi?

File: devel/hg-git/Portfile
Comment (lbschenkel) [inline devel/hg-git/Portfile:43]: Can you add comments here linking to the source for each patch (a PR, etc.)?

File: python/py-mercurial_extension_utils/Portfile
Comment (lbschenkel) [inline python/py-mercurial_extension_utils/Portfile:30]: Why `39` again?

**Review by lbschenkel — CHANGES_REQUESTED** [review]
I think you have a single commit doing too much: bumping the ports and changing the default Python version should not be in the very same commit. If you want to change to Python 3.9, have a separate commit that does only that.

File: python/py-mercurial_extension_utils/Portfile
Comment (danchr) [reply python/py-mercurial_extension_utils/Portfile:18]: The build failed due to changed distfile checksums, so I figured getting the binary fro PyPI was more stable.

Comment (reneeotten) [reply python/py-mercurial_extension_utils/Portfile:18]: I still think one should not mix and match in this case; pick one approach (i.e., either gitlab or PyPI) and stick with it. If there is a checksum mismatch you can treat it as [a stealth update](https://trac.macports.org/wiki/PortfileRecipes#stealth-updates).

**Comment by reneeotten** [comment]
ping @danchr - do you want to finish this up or should this PR be closed?

**Comment by danchr** [comment]
> ping @danchr - do you want to finish this up or should this PR be closed?

Sorry 'bout the delay; should be much better now.

File: devel/hg-evolve/Portfile
Comment (reneeotten) [inline devel/hg-evolve/Portfile:50]: does this actually allow to run the tests? I would think you should at least set `test.run yes` and then ideally use the typical `test.cmd` and related things to run the actual tests instead of doing a system call.

File: devel/hg-keyring/Portfile
Comment (reneeotten) [inline devel/hg-keyring/Portfile:11]: this should be `categories-prepend`

File: devel/hg-keyring/Portfile
Comment (reneeotten) [inline devel/hg-keyring/Portfile:22]: please add `homepage` field

File: devel/hg-keyring/Portfile
Comment (reneeotten) [inline devel/hg-keyring/Portfile:9]: you define `name` twice

File: devel/hg-keyring/Portfile
Comment (reneeotten) [inline devel/hg-keyring/Portfile:45]: does `mercurial` need to be both `depends_lib` and `depends_run` - that seems redundant, right?

**Comment by szhorvat** [comment]
@danchr Perhaps this could update to mercurial 5.7.1 now?

File: devel/git-remote-hg/Portfile
Comment (lbschenkel) [inline devel/git-remote-hg/Portfile:35]: Is this here because the former declarations no longer work due to `python` PortGroup?

File: devel/hg-git/Portfile
Comment (lbschenkel) [inline devel/hg-git/Portfile:7]: Are you sure that we should revert to pypi? The reason that we were building from source from the canonical location and not always using pypi is because this project had a history of having very infrequent releases, and the extension would break on newer versions of Mercurial, and fixes would be committed in `master` but not released for a long time.

**Review by lbschenkel — COMMENTED** [review]
Nice work. I have a couple of comments/questions/observations.

File: devel/git-remote-hg/Portfile
Comment (danchr) [reply devel/git-remote-hg/Portfile:35]: Well, it's rather that they weren't needed; using the Python port group ensures that the Python modules are correctly installed, and this ensures that we still get the documentation. To be honest, I don't remember whether it didn't actually work; if you'd prefer to keep the previous style, I can look into it?

File: devel/hg-git/Portfile
Comment (danchr) [reply devel/hg-git/Portfile:7]: I was made co-maintainer of hg-git not long ago, and I actually pushed the most recent release to PyPI myself. So if it's out of date sometime in the future, please do complain 🙂

File: devel/git-remote-hg/Portfile
Comment (lbschenkel) [reply devel/git-remote-hg/Portfile:35]: This is a small detail and I don't want to add roadblocks to merge this PR, however if the "declarative style" works I think we should favour that because that's the idiomatic way and it leads to more maintainable ports.

File: devel/hg-git/Portfile
Comment (lbschenkel) [reply devel/hg-git/Portfile:7]: All right then. :-) Good to hear.

File: devel/git-remote-hg/Portfile
Comment (danchr) [reply devel/git-remote-hg/Portfile:35]: My general goal was to make it easier to keep the Python version in sync with Mercurial by using the Python port group. I _think_ the issue was that using that group with another build system is tricky, and it's much easier to just build the documentation separately.

Comment (lbschenkel) [reply devel/git-remote-hg/Portfile:35]: OK. If you tried and it didn't work, or it required too much fiddling, then it is fine as it is. I just wanted to understand the reasoning for the change.

**Review by lbschenkel — APPROVED** [review]

**Review by reneeotten — APPROVED** [review]

**Comment by lbschenkel** [comment]
Two approvals and green build, went ahead and merged.

**Comment by danchr** [comment]
> Two approvals and green build, went ahead and merged.

Thanks!

**Pull request merged**
